src/psyclone/gocean1p0.py

```diff
@@ -880,6 +881,222 @@ def backward_dependence(self):
         return None
 
 
+class GOStencil(object):
+    '''GOcean 1.0 stencil information for a kernel argument as obtained by
+    parsing the kernel meta-data. The expected structure of the
+    metadata and its meaning is provided in the description of the
+    load method
+
+    '''
+    def __init__(self):
+        ''' Set up any internal variables. '''
+        self._has_stencil = None
+        self._stencil = [[0 for _ in range(3)] for _ in range(3)]
+        self._name = None
+        self._initialised = False
+
+    def load(self, stencil_info, kernel_name):
+        '''Take parsed stencil metadata information, check it is valid and
+        store it in a convenient form. The kernel_name argument is
+        only used to provide the location if there is an error.
+
+        The stencil information should either be a name which
+        indicates a particular type of stencil or in the form
+        stencil(xxx,yyy,zzz) which explicitly specifies a stencil
+        shape where xxx, yyy and zzz are triplets of integers
+        indicating whether there is a stencil access in a particular
+        direction and the depth of that access. For example:
+
+        stencil(010,  !   N
+                212,  !  W E
+                010)  !   S
+
+        indicates that there is a stencil access of depth 1 in the
+        "North" and "South" directions and and stencil access of depth
+        1 and 2 in the "East" and "West" directions. The value at the
```

Comment:
Repeated "and" and both the East and West directions have a depth of 2.

Reply:
Well spotted